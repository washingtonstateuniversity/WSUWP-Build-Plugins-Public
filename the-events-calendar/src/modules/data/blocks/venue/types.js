/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_STORE } from '@moderntribe/events/data/utils';

export const SET_INITIAL_STATE = `${ PREFIX_EVENTS_STORE }/VENUE_BLOCK/SET_INITIAL_STATE`;

export const SET_VENUE = `${ PREFIX_EVENTS_STORE }/SET_VENUE`;

export const TOGGLE_VENUE_MAP = `${ PREFIX_EVENTS_STORE }/TOGGLE_VENUE_MAP`;
export const SET_VENUE_MAP = `${ PREFIX_EVENTS_STORE }/SET_VENUE_MAP`;

export const TOGGLE_VENUE_MAP_LINK = `${ PREFIX_EVENTS_STORE }/TOGGLE_VENUE_MAP_LINK`;
export const SET_VENUE_MAP_LINK = `${ PREFIX_EVENTS_STORE }/SET_VENUE_MAP_LINK`;
