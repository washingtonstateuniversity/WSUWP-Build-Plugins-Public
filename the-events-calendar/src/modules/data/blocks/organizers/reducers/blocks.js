/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import * as types from './../types';
import block from './block';

export const byId = ( state = {}, action ) => {
	switch ( action.type ) {
		case types.ADD_ORGANIZER_BLOCK:
		case types.REMOVE_ORGANIZER_BLOCK:
			return {
				...state,
				[ action.payload.id ]: block( state[ action.payload.id ], action ),
			};
		default:
			return state;
	}
};

export const allIds = ( state = [], action ) => {
	switch ( action.type ) {
		case types.ADD_ORGANIZER_BLOCK:
			return [ ...state, action.payload.organizer ];
		case types.REMOVE_ORGANIZER_BLOCK:
			return [ ...state ].filter( ( organizer ) => organizer !== action.payload.organizer );
		default:
			return state;
	}
};

export default combineReducers( {
	byId,
	allIds,
} );
