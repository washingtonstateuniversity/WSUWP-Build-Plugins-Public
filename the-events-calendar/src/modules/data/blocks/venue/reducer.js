/**
 * Internal dependencies
 */
import * as types from './types';
import { editorDefaults } from '@moderntribe/common/utils/globals';

export const DEFAULT_STATE = {
	venue: editorDefaults().venue ? editorDefaults().venue : 0,
	showMap: true,
	showMapLink: true,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_VENUE:
			return {
				...state,
				venue: action.payload.venue,
			};
		case types.TOGGLE_VENUE_MAP:
			return {
				...state,
				showMap: ! state.showMap,
			};
		case types.SET_VENUE_MAP:
			return {
				...state,
				showMap: action.payload.showMap,
			};
		case types.TOGGLE_VENUE_MAP_LINK:
			return {
				...state,
				showMapLink: ! state.showMapLink,
			};
		case types.SET_VENUE_MAP_LINK:
			return {
				...state,
				showMapLink: action.payload.showMapLink,
			};
		default:
			return state;
	}
};
