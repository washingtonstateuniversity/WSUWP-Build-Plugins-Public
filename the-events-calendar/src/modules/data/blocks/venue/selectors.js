/**
 * External dependencies
 */
import { createSelector } from 'reselect';

export const venueBlockSelector = ( state ) => state.events.blocks.venue;
export const getVenue = createSelector(
	[ venueBlockSelector ],
	( block ) => block.venue,
);

export const getshowMapLink = createSelector(
	[ venueBlockSelector ],
	( block ) => block.showMapLink,
);

export const getshowMap = createSelector(
	[ venueBlockSelector ],
	( block ) => block.showMap,
);
