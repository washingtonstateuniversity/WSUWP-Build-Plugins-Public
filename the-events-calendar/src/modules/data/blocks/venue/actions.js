/**
 * Internal dependencies
 */
import * as types from './types';
import { DEFAULT_STATE } from './reducer';

export const setVenue = ( id ) => ( {
	type: types.SET_VENUE,
	payload: {
		venue: id,
	},
} );

export const removeVenue = () => ( {
	type: types.SET_VENUE,
	payload: {
		venue: undefined,
	},
} );

export const toggleVenueMap = () => ( { type: types.TOGGLE_VENUE_MAP } );

export const setShowMap = ( showMap ) => ( {
	type: types.SET_VENUE_MAP,
	payload: {
		showMap,
	},
} );

export const toggleVenueMapLink = () => ( { type: types.TOGGLE_VENUE_MAP_LINK } );

export const setShowMapLink = ( showMapLink ) => ( {
	type: types.SET_VENUE_MAP_LINK,
	payload: {
		showMapLink,
	},
} );

export const setInitialState = ( props ) => ( {
	type: types.SET_INITIAL_STATE,
	payload: props,
} );

