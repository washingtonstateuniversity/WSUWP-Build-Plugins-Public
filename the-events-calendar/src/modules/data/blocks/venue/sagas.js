/**
 * External Dependencies
 */
import { put, all, takeEvery } from 'redux-saga/effects';

/**
 * Internal dependencies
 */
import * as types from './types';
import { DEFAULT_STATE } from './reducer';
import * as actions from './actions';

export function* setInitialState( action ) {
	const { get } = action.payload;
	const venue = get( 'venue' ) ? get( 'venue' ) : DEFAULT_STATE.venue;

	yield all( [
		put( actions.setVenue( venue ) ),
		put( actions.setShowMap( get( 'showMap', DEFAULT_STATE.showMap ) ) ),
		put( actions.setShowMapLink( get( 'showMapLink', DEFAULT_STATE.showMapLink ) ) ),
	] );
}

export default function* watchers() {
	yield takeEvery( types.SET_INITIAL_STATE, setInitialState );
}
