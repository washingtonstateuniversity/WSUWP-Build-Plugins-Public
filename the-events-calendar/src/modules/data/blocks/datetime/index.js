/**
 * Internal dependencies
 */
import reducer from './reducer';
import * as types from './types';
import * as actions from './actions';
import * as thunks from './thunks';
import * as selectors from './selectors';
import sagas from './sagas';

export default reducer;
export { types, actions, thunks, selectors, sagas };
