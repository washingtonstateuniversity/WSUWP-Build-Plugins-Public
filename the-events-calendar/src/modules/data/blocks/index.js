/**
 * Internal dependencies
 */
import * as datetime from './datetime';

export { datetime };
export { default } from './reducer';
