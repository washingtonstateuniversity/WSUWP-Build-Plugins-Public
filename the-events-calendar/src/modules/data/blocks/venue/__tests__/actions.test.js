/**
 * External dependencies
 */
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
/**
 * Internal dependencies
 */
import { actions } from '@moderntribe/events/data/blocks/venue';

const middlewares = [ thunk ];
const mockStore = configureStore( middlewares );

describe( '[STORE] - Venue actions', () => {
	test( 'action to set initial state', () => {
		expect( actions.setInitialState() ).toMatchSnapshot();
	} );
	test( 'action to set the venue', () => {
		expect( actions.setVenue( 99 ) ).toMatchSnapshot();
	} );

	test( 'action to set the showMap', () => {
		expect( actions.setShowMap( false ) ).toMatchSnapshot();
		expect( actions.setShowMap( true ) ).toMatchSnapshot();
	} );

	test( 'action to set the showMapLink', () => {
		expect( actions.setShowMapLink( true ) ).toMatchSnapshot();
		expect( actions.setShowMapLink( false ) ).toMatchSnapshot();
	} );

	test( 'action to set the venue removal', () => {
		expect( actions.removeVenue() ).toMatchSnapshot();
	} );

	test( 'action to toggle the venue map', () => {
		expect( actions.toggleVenueMap() ).toMatchSnapshot();
	} );

	test( 'action to toggle the venueMapLink', () => {
		expect( actions.toggleVenueMapLink() ).toMatchSnapshot();
	} );
} );
