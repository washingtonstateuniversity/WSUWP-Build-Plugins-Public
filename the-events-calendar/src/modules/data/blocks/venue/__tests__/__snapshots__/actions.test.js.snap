// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - Venue actions action to set initial state 1`] = `
Object {
  "payload": undefined,
  "type": "@@MT/EVENTS/VENUE_BLOCK/SET_INITIAL_STATE",
}
`;

exports[`[STORE] - Venue actions action to set the showMap 1`] = `
Object {
  "payload": Object {
    "showMap": false,
  },
  "type": "@@MT/EVENTS/SET_VENUE_MAP",
}
`;

exports[`[STORE] - Venue actions action to set the showMap 2`] = `
Object {
  "payload": Object {
    "showMap": true,
  },
  "type": "@@MT/EVENTS/SET_VENUE_MAP",
}
`;

exports[`[STORE] - Venue actions action to set the showMapLink 1`] = `
Object {
  "payload": Object {
    "showMapLink": true,
  },
  "type": "@@MT/EVENTS/SET_VENUE_MAP_LINK",
}
`;

exports[`[STORE] - Venue actions action to set the showMapLink 2`] = `
Object {
  "payload": Object {
    "showMapLink": false,
  },
  "type": "@@MT/EVENTS/SET_VENUE_MAP_LINK",
}
`;

exports[`[STORE] - Venue actions action to set the venue 1`] = `
Object {
  "payload": Object {
    "venue": 99,
  },
  "type": "@@MT/EVENTS/SET_VENUE",
}
`;

exports[`[STORE] - Venue actions action to set the venue removal 1`] = `
Object {
  "payload": Object {
    "venue": undefined,
  },
  "type": "@@MT/EVENTS/SET_VENUE",
}
`;

exports[`[STORE] - Venue actions action to toggle the venue map 1`] = `
Object {
  "type": "@@MT/EVENTS/TOGGLE_VENUE_MAP",
}
`;

exports[`[STORE] - Venue actions action to toggle the venueMapLink 1`] = `
Object {
  "type": "@@MT/EVENTS/TOGGLE_VENUE_MAP_LINK",
}
`;
